src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs

```diff
@@ -644,6 +657,18 @@ private static bool InRange(int ch, int start, int end)
         /// <returns>True if the input buffer was fully encoded into the output buffer, otherwise false.</returns>
         public static unsafe bool TryEncode(ReadOnlySpan<char> utf16, Span<byte> utf8, out int charactersConsumed, out int bytesWritten)
         {
+            if (utf16.IsEmpty)
```

Comment:
What about if the output buffer is empty (and input buffer is not)? pTarget is null and we don't return from the method if available = 0 (we just break from the fast loop and we enter the slow loop), which causes this to execute.
```C#
*pTarget = (byte)chd;
```
https://github.com/dotnet/corefxlab/blob/master/src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs#L901